platform/platform-filter-delegate/src/main/java/org/codice/ddf/platform/filter/delegate/DelegateServletFilter.java

```diff
@@ -14,104 +14,286 @@
+                bundleContext));
+        keysOfInitializedSecurityFilters.add(getFilterKey(securityFilterServiceReference,
+                bundleContext));
+        LOGGER.debug("Initialized SecurityFilter {}", filterName);
+    }
+
+    public void removeSecurityFilter(
+            final ServiceReference<SecurityFilter> securityFilterServiceReference) {
+        final BundleContext bundleContext = getContext();
+
+        if (bundleContext != null) {
+            // unmark the SecurityFilter as initialized so that it can be re-initialized if the SecurityFilter is registered again
+            keysOfInitializedSecurityFilters.remove(getFilterKey(securityFilterServiceReference,
+                    bundleContext));
+            bundleContext.getService(securityFilterServiceReference)
+                    .destroy();
+        } else {
+            LOGGER.warn(
+                    "Unable to remove SecurityFilter {}. Try restarting the system or turning up logging to monitor current SecurityFilters.");
+        }
+    }
+
+    @Nonnull
+    private static String getFilterKey(
+            final ServiceReference<SecurityFilter> securityFilterServiceReference,
+            final BundleContext bundleContext) {
+        return getFilterName(securityFilterServiceReference, bundleContext);
+    }
+
+    /**
+     * This logic to get the filter name from a {@link ServiceReference<Filter>} is copied from
+     * {@link org.ops4j.pax.web.extender.whiteboard.internal.tracker.ServletTracker#createWebElement(ServiceReference, javax.servlet.Servlet)}.
+     * See the pax-web Whiteboard documentation and
+     * {@link org.osgi.service.http.whiteboard.HttpWhiteboardConstants#HTTP_WHITEBOARD_FILTER_NAME}
+     * for how to configure {@link Filter} services with a filter name.
+     */
+    @Nonnull
+    private static String getFilterName(
+            ServiceReference<SecurityFilter> securityFilterServiceReference,
+            BundleContext bundleContext) {
+        final String httpWhiteboardFilterName = "osgi.http.whiteboard.filter.name";
+        final String filterNameFromTheServiceProperty = getStringProperty(
+                securityFilterServiceReference,
+                httpWhiteboardFilterName);
+        // If this service property is not specified, the fully qualified name of the service object's class is used as the servlet filter name.
+        if (StringUtils.isBlank(filterNameFromTheServiceProperty)) {
+            return bundleContext.getService(securityFilterServiceReference)
+                    .getClass()
+                    .getCanonicalName();
+        } else {
+            return filterNameFromTheServiceProperty;
+        }
+    }
+
+    private static String getStringProperty(ServiceReference<?> serviceReference, String key) {
+        Object value = serviceReference.getProperty(key);
+        if (value != null && !(value instanceof String)) {
+            LOGGER.warn("Service property [" + key + "] value must be a String");
```

Comment:
❗️ To rephrase Sonarqube...use variable interpolation instead of concatenation for your log message.